main/src/library/Ping.flix

```diff
@@ -0,0 +1,71 @@
+/*
+ *  Copyright 2025 Cade Lueker
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+///
+/// The effect used to determine if an address is reachable.
+///
+eff Ping {
+    def isReachable(host: String, timeout: Int32, u: TimeUnit): Bool
+}
+
+mod Ping {
+    use IoError.ErrorKind
+    use IoError.IoError
+
+    import java.net.InetAddress
+    import java.io.IOException
+    import java.net.UnknownHostException
+    import java.lang.IllegalArgumentException
+
+    ///
+    /// Handles the `Ping` effect of the given function `f`.
+    ///
+    /// In other words, re-interprets the `Ping` effect using the `Net` and `IO` effects.
+    ///
+    pub def handle(f: a -> b \ ef): a -> Result[IoError, b] \ (ef - Ping) + {Net, IO} = x ->
+        run {
+            Ok(f(x))
+        } with handler Ping {
+            def isReachable(host, timeout, k) = {
+                try {
+                    let timeoutMs = match u {
+                           case TimeUnit.Days => timeout * 86400000,
+                           case TimeUnit.Hours => timeout * 3600000,
+                           case TimeUnit.Microseconds => timeout / 1000,
+                           case TimeUnit.Milliseconds => timeout,
+                           case TimeUnit.Minutes=> timeout * 6000,
+                           case TimeUnit.Nanoseconds=> timeout / 1000000,
+                           case TimeUnit.Seconds => timeout * 1000
```

Comment:
We will probably want to move the conversion somewhere else so we can reuse it, maybe into TimeUnit module?

Reply:
Also, I would avoid magic constants but instead have the multiplications there.

Oh and use _ for thousand separator, e.g. 1_000